0007-configuration-management-extension/README.md

```diff
@@ -33,51 +37,75 @@ It is intended to manage endpoint configuration distribution.
     Possible solutions:
     - Send only updates. (e.g., using JSON Patch format, defined in [RFC 6902](https://tools.ietf.org/html/rfc6902).)
 
-## Use cases
+4. _Endpoint can actively reject a configuration._
+
+   Examples are:
+   - the endpoint is unable to process the configuration.
+   - the configuration is ill-formated, or pre-conditions are not met.
+
+   Possible solutions:
+   - Send the status result of the configuration application (either success or failure) back to the server.
+
+# Use cases
 
-### UC1: Configuration request
+## UC1: Configuration request
 Configuration delivery by request. The endpoint should be able to receive latest configuration from CMX by request.
 
-### UC2: Configuration push
+## UC2: Configuration push
 Configuration delivery is initiated by the server. The endpoint should receive latest configuration when it connects to the server if this configuration wasn't applied yet.
 
-## Design
+# Design
 The extension follows client-initiated request/response pattern described in 1/KP, which nicely maps to both MQTT and CoAP.
 
 There are two resources provided by the extension.
 
 - `/<endpoint_token>/config/json` is used for subscribing and receiving a config.
 - `/<endpoint_token>/applied/json` is used for notifiying server of the currently active configuration.
 
-### Configuration identifier
+## Configuration identifier
 Configuration identifier is an opaque string, which uniquely identifies the configuration. The client SHOULD NOT assume the nature of identifiers.
 
-### Absent configuration
+## Absent configuration
```

Comment:
I kind of agree with you but would like error responses to signify either client or server error. (So you only see them during debugging and can expect a good application never sees an error response and does not care about parsing them.)

Absent configuration is neither client nor server error.

I need to think more about that.